native/src/hooks/useLocation.ts

```diff
@@ -0,0 +1,141 @@
+import Geolocation, { GeolocationError, GeolocationResponse } from '@react-native-community/geolocation'
+import { useCallback, useEffect, useState } from 'react'
+import { openSettings, RESULTS } from 'react-native-permissions'
+import SystemSetting from 'react-native-system-setting'
+
+import { checkLocationPermission, requestLocationPermission } from '../utils/LocationPermissionManager'
+
+type SuccessfulLocationState = { status: 'ready' }
+type UnavailableLocationState = {
+  status: 'unavailable'
+  message: 'noPermission' | 'notAvailable' | 'timeout' | 'loading'
+}
+
+type LocationStateType = SuccessfulLocationState | UnavailableLocationState
+
+export type LocationType = [number, number]
+
+const locationStateOnError = (error: GeolocationError): UnavailableLocationState => {
+  if (error.code === 1) {
+    return {
+      status: 'unavailable',
+      message: 'noPermission'
+    }
+  } else if (error.code === 2) {
+    return {
+      status: 'unavailable',
+      message: 'notAvailable'
+    }
+  } else {
+    return {
+      status: 'unavailable',
+      message: 'timeout'
+    }
```

Comment:
Is this the inteface? 
https://developer.mozilla.org/en-US/docs/Web/API/GeolocationPositionError/code

Another value then 3 is unexpcted. So perhaps better we use another if here (or switch case) with "3" (or TIMEOUT, see other comment). In the else we could put something like "status unknown" or event throw an error.

Reply:
Probably yes. But the docs you referenced is from web, therefore I'll use their interface